Main.cs

```diff
@@ -612,6 +628,7 @@ public static void Run(string[] _args)
                             }
                         }
                     }
+                    return;

```

Comment:
For all of these return calls.  Is it worth doing something like the following.  Make it easier to follow what Main is doing?
![image](https://cloud.githubusercontent.com/assets/6966179/5992136/f35d6c38-a9e8-11e4-9cd7-55d269db8c8c.png)

Not sure if this is within scope of the PR